core/model.py

```diff
@@ -65,7 +65,11 @@ def _post_process(self, x):
     def _load_assets(self, path):
         pass
 
-    def predict(self, x, model):
+    def predict(self, args):
```

Comment:
I think we should move `_pre_process()` out of this so it can be refactored to `_predict()`.